apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -124,33 +123,19 @@ func MakeProvisionerAPI(stdCtx context.Context, ctx facade.ModelContext) (*Provi
 		return authorizer.AuthOwner, nil
 	}
 	st := ctx.State()
-	model, err := st.Model()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
 	serviceFactory := ctx.ServiceFactory()
 
-	configGetter := stateenvirons.EnvironConfigGetter{
-		Model:             model,
-		CloudService:      serviceFactory.Cloud(),
-		CredentialService: serviceFactory.Credential(),
-	}
-
 	modelInfoService := serviceFactory.ModelInfo()
 	modelInfo, err := modelInfoService.GetModelInfo(stdCtx)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	isCaasModel := modelInfo.Type == coremodel.CAAS
 
-	var env storage.ProviderRegistry
-	if isCaasModel {
-		env, err = stateenvirons.GetNewCAASBrokerFunc(caas.New)(model, serviceFactory.Cloud(), serviceFactory.Credential())
-	} else {
-		env, err = environs.GetEnviron(stdCtx, configGetter, environs.New)
-	}
+	environGetter := facade.ProviderRunner[environs.Environ](ctx)
+	env, err := environGetter(stdCtx)
```

Comment:
Essentially an environ may or may not implement extra interface methods such as LXDProfiler. With the `providerFactory.GetProvider`, we would have to take the output and do a type assertion to check if it has the extra method. 

The `facade.ProviderRunner` is a convenience method that does this for us. Internally it will call `GetProvider`, and then do a type assertion with the specified generic type argument. If the required interface is not implemented by this specific environ, then it automatically returns a NotSupported error for us. It's similar to what we do in the domain packages with `providertracker.ProviderRunner`.